tests/dual-farming/single-reward.ts

```diff
@@ -0,0 +1,219 @@
+import * as anchor from "@project-serum/anchor";
+import {Token, TOKEN_PROGRAM_ID} from "@solana/spl-token";
+import {LAMPORTS_PER_SOL} from "@solana/web3.js";
+import {DualFarming} from "../../target/types/dual_farming";
+import {
+  getPoolPda,
+  getRewardAVaultPda,
+  getRewardBVaultPda,
+  getStakingVaultPda,
+  getUserPda,
+} from "./utils";
+import {Program} from "@project-serum/anchor";
+
+const provider = anchor.AnchorProvider.env();
+anchor.setProvider(provider);
+
+const program = anchor.workspace.DualFarming as Program<DualFarming>;
+
+const BASE_KEYPAIR = anchor.web3.Keypair.generate();
+const ADMIN_KEYPAIR = anchor.web3.Keypair.generate();
+const USER_KEYPAIR = anchor.web3.Keypair.generate();
+const FUNDER_KEYPAIR = anchor.web3.Keypair.generate();
+const REWARD_DURATION = new anchor.BN(5);
+const TOKEN_DECIMAL = 3;
+const TOKEN_MULTIPLIER = 10 ** TOKEN_DECIMAL;
+
+
+describe("dual-farming with single reward", () => {
+  let stakingMint: anchor.web3.PublicKey = null;
+  let rewardMint: anchor.web3.PublicKey = null;
+
+  let stakingToken: Token = null;
+  let rewardToken: Token = null;
+
+  let userStakingATA: anchor.web3.PublicKey = null;
+  let userRewardATA: anchor.web3.PublicKey = null;
+
+  let adminStakingATA: anchor.web3.PublicKey = null;
+  let adminRewardATA: anchor.web3.PublicKey = null;
+
+  before(async () => {
+    let sig = await program.provider.connection.requestAirdrop(
+      ADMIN_KEYPAIR.publicKey,
+      100 * LAMPORTS_PER_SOL
+    );
+    await program.provider.connection.confirmTransaction(sig);
+
+    sig = await program.provider.connection.requestAirdrop(
+      USER_KEYPAIR.publicKey,
+      100 * LAMPORTS_PER_SOL
+    );
+    await program.provider.connection.confirmTransaction(sig);
+
+    stakingToken = await Token.createMint(
+      program.provider.connection,
+      ADMIN_KEYPAIR,
+      ADMIN_KEYPAIR.publicKey,
+      null,
+      TOKEN_DECIMAL,
+      TOKEN_PROGRAM_ID
+    );
+    stakingMint = stakingToken.publicKey;
+    userStakingATA = await stakingToken.createAssociatedTokenAccount(
+      USER_KEYPAIR.publicKey
+    );
+    adminStakingATA = await stakingToken.createAssociatedTokenAccount(
+      ADMIN_KEYPAIR.publicKey
+    );
+
+    rewardToken = await Token.createMint(
+      program.provider.connection,
+      ADMIN_KEYPAIR,
+      ADMIN_KEYPAIR.publicKey,
+      null,
+      TOKEN_DECIMAL,
+      TOKEN_PROGRAM_ID
+    );
+    rewardMint = rewardToken.publicKey;
+    userRewardATA = await rewardToken.createAssociatedTokenAccount(
+      USER_KEYPAIR.publicKey
+    );
+    adminRewardATA = await rewardToken.createAssociatedTokenAccount(
+      ADMIN_KEYPAIR.publicKey
+    );
+
+
+    console.log("Program ID: ", program.programId.toString());
+    console.log("Wallet: ", provider.wallet.publicKey.toString());
+
+
+  });
+
+  it("should initialize dual-farming with single reward pool", async () => {
```

Comment:
Could you comment details how to setup single farming?

Reply:
Phi had fixed it

Reply:
Phi had fixed it